pkg/metrics/filters/filters_test.go

```diff
@@ -0,0 +1,253 @@
+package filters
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/prometheus/client_golang/prometheus"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+)
+
+var _ = Describe("manger.Manager", func() {
+	Describe("Start", func() {
+		Context("should start serving metrics with https and authn/authz", func() {
+			var srv metricsserver.Server
+			var defaultServer metricsDefaultServer
+			var opts manager.Options
+			var httpClient *http.Client
+
+			BeforeEach(func() {
+				srv = nil
+				opts = manager.Options{
+					Metrics: metricsserver.Options{
+						BindAddress:    ":0",
+						SecureServing:  true,
+						FilterProvider: WithAuthenticationAndAuthorization,
+					},
+					NewMetricsServer: func(options metricsserver.Options, config *rest.Config, httpClient *http.Client) (metricsserver.Server, error) {
+						var err error
+						srv, err = metricsserver.NewServer(options, config, httpClient)
+						if srv != nil {
+							defaultServer = srv.(metricsDefaultServer)
+						}
+						return srv, err
+					},
+				}
+				httpClient = &http.Client{Transport: &http.Transport{
+					TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, //nolint:gosec
+				}}
+			})
+
+			It("should serve metrics in its registry", func() {
```

Comment:
Looks like we lost some test cases. Was this intentional?

Reply:
I had thought I captured the authz specific ones here and the general metrics ones left in manager/. but maybe not, let me double check

Reply:
Just compared and saw that we lost a few. But they were the easiest ones. Not sure how much additional code paths they cover. If it's nothing new definitely fine to not add them again